amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/ShardSyncer.java

```diff
@@ -71,7 +68,7 @@
      * @throws KinesisClientLibIOException
      */
     // CHECKSTYLE:OFF CyclomaticComplexity
-    public static synchronized void checkAndCreateLeasesForNewShards(@NonNull final ShardDetector shardDetector,
+    public synchronized void checkAndCreateLeasesForNewShards(@NonNull final ShardDetector shardDetector,
```

Comment:
This seems like a breaking change. A better way would be to deprecate the existing method and creating a new non-static method.

Reply:
You are right, modified.